platforms/ide/tooling-api/src/main/java/org/gradle/tooling/internal/protocol/InternalObjectRelay.java

```diff
@@ -17,11 +17,18 @@
 package org.gradle.tooling.internal.protocol;
 
 /**
+ * <p>DO NOT CHANGE THIS INTERFACE - it is part of the cross-version protocol.
```

Comment:
❔   We use this `DO NOT CHANGE THIS INTERFACE` phrase with almost all protocol interfaces. But isn't it correct, that this interface can be extended with new methods, and it will not be a breaking change?

Reply:
These interfaces are implemented by and used from different versions of the tooling API. So, if we add a method to one of these interfaces, then the old side will not implement the new method and the call from the new side will fail with an "abstract method" exception.

Maybe we could switch to using some structural typing for these interfaces as well like we do for the parameter types, which should mean we can safely add methods, but I don't want to change the pattern as part of this work.